### PR TITLE
Add Homebrew test-bot workflow

### PR DESCRIPTION
**Pull request by arbourd**
Tests to see if the `subo` Formula will run. Will install and run tests for the binary.

### CODE REVIEW

File: .github/actions/formula.tmpl [hunk 1]
```diff
@@ -1,10 +1,9 @@
 class Subo < Formula
-  desc "The Suborbital CLI for building and deploying WebAssembly-based projects."
+  desc "Suborbital CLI for building and deploying WebAssembly-based projects"
   homepage "https://suborbital.dev"
-  head "https://github.com/suborbital/subo.git"
   url "https://download.suborbital.network/subo/brew/<<<tag>>>"
+  head "https://github.com/suborbital/subo.git"
   sha256 "<<<sha256>>>"
-  version "<<<tag>>>"
```
Comment (arbourd) [inline .github/actions/formula.tmpl:7]: Homebrew doesn't like it when it infers version and you provide it

File: .github/actions/formula.tmpl [hunk 2]
```diff
@@ -14,6 +13,6 @@ class Subo < Formula
   end
 
   test do
-    system "#{bin}/subo --version"
+    assert_match "/#{version}/", shell_output("#{bin}/subo --version")
```
Comment (arbourd) [inline .github/actions/formula.tmpl:16]: This calls the binary but ensures that it spits out the version it says it is.

File: .github/actions/formula.tmpl (hunk 1 above)
Comment (javorszky) [reply .github/actions/formula.tmpl:7]: TIL, okay

File: .github/workflows/formula.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  push:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: macos-latest
+    env:
+      HOMEBREW_GITHUB_ACTIONS: 1
+      HOMEBREW_NO_ANALYTICS: 1
```
Comment (javorszky) [inline .github/workflows/formula.yml:14]: yay!

**Review by javorszky — APPROVED** [review]
Looks good, though that formula error the github actions flagged up needs fixing. Not sure why it thinks it's a cask rather than a formula.

File: .github/actions/formula.tmpl (hunk 1 above)
Comment (arbourd) [reply .github/actions/formula.tmpl:7]: ya very opinionated, lol.

**Comment by arbourd** [comment]
@javorszky I think that's because I'm using the tool in a way that doesn't jive normally with their stuff? It will change with the next `subo` release tho so I'll merge it for now and keep an eye on it.

**Pull request merged**